source/common/tcp_proxy/tcp_proxy.h

```diff
@@ -111,39 +145,65 @@ class Config {
   }
   UpstreamDrainManager& drainManager();
   SharedConfigSharedPtr sharedConfig() { return shared_config_; }
-  const Router::MetadataMatchCriteria* metadataMatchCriteria() {
+  const Router::MetadataMatchCriteria* metadataMatchCriteria() const {
     return cluster_metadata_match_criteria_.get();
   }
   const Network::HashPolicy* hashPolicy() { return hash_policy_.get(); }
 
 private:
-  struct Route {
-    Route(const envoy::config::filter::network::tcp_proxy::v2::TcpProxy::DeprecatedV1::TCPRoute&
-              config);
+  struct TCPRoute : public RouteEntry, public Route {
```

Comment:
nit: let's call this `RouteImpl` to match the pattern used in the http connection manager and elsewhere.